test/unit/jsTest.js

```diff
@@ -0,0 +1,234 @@
+          showWarnings: false,
+          params: {}
+        },
+        output: '.build/js',
+        sourceDir: 'js',
+        whitelist: ['a.js', 'c.js']
+      }
+    }, 'initServer')
+    // create a fork of it and run it
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prod'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', () => {
+      // test to see that only the whitelisted file was compiled
+      const compiledJS = path.join(path.join(appDir, 'statics', '.build', 'js'))
+      const compiledJSArray = klawSync(compiledJS)
+      compiledJSArray.forEach((file) => {
+        let test = pathOfWhiteListedFiles.includes(file.path)
+        assert.equal(test, true)
+      })
+      testApp.kill()
+      done()
+    })
+  })
+
+  it.skip('should minify all files except for those that are blacklisted', function (done) {
+    // change the data that would be put into the static files
+    staticJSFiles[0] = `var a = function() { return 1 + 2}`
+    staticJSFiles[1] = `var b = function(multin) { return multin * 4}`
+    staticJSFiles[2] = `var c = function(name) {console.log("Hello " + name)}`
+
+    generateStaticFolder()
+    // get the buffer of the static files
+    let staticJSFilesA = fs.readFileSync(pathsOfStaticJS[0], 'utf8')
+    let staticJSFilesB = fs.readFileSync(pathsOfStaticJS[1], 'utf8')
+    let staticJSFilesC = fs.readFileSync(pathsOfStaticJS[2], 'utf8')
+
+    generateTestApp({
+      appDir: appDir,
+      generateFolderStructure: true,
+      suppressLogs: {
+        httpLogs: true,
+        rooseveltLogs: true,
+        rooseveltWarnings: true
+      },
+      js: {
+        compiler: {
+          nodeModule: 'roosevelt-uglify',
+          showWarnings: false,
+          params: {}
+        },
+        output: '.build/js',
+        sourceDir: 'js',
+        whitelist: [],
+        blacklist: ['c.js']
+      }
+    }, 'initServer')
+
+    // create a fork of it and run it
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prod'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', () => {
+      // grab all the compiled file's info
+      let compiledJSFilesA = fs.readFileSync(pathsOfCompiledJS[0], 'utf8')
+      let compiledJSFilesB = fs.readFileSync(pathsOfCompiledJS[1], 'utf8')
+      let compiledJSFilesC = fs.readFileSync(pathsOfCompiledJS[2], 'utf8')
+      // see if the buffer from the compiled is the same as their static counterpart
+      let test1 = staticJSFilesA === compiledJSFilesA
+      let test2 = staticJSFilesB === compiledJSFilesB
+      let test3 = staticJSFilesC === compiledJSFilesC
+      assert.equal(test1, false)
+      assert.equal(test2, false)
+      assert.equal(test3, true)
+      testApp.kill()
+      done()
+    })
+  })
+
+  it('should make the output compiled folder with the new name and put all the compiled JS in it', function (done) {
+    // make everything back to normal
+    staticJSFiles[0] = 'var a = 7'
```

Comment:
There should be no reason to change these variables inside of tests. They should be specified in the original array and stay the same for the entire suite.